include/aspect/utilities.h

```diff
@@ -955,6 +957,63 @@ namespace aspect
                               const SymmetricTensor<2,dim> &dviscosities_dstrain_rate,
                               const double safety_factor);
 
+    /**
+     * A class that represents a binary operator between two doubles. The type of
+     * operation is specified on construction time, and can be checked later
+     * by using the operator ==. The operator () executes the operation on two
+     * double parameters and returns the result. This class is helpful for
+     * user specified operations that are not known at compile time.
+     */
+    class Operator
+    {
+      public:
+        /**
+         * An enum of supported operations.
+         */
+        enum operation
+        {
+          add,
+          subtract,
+          minimum,
+          maximum,
+          uninitialized
```

Comment:
I should have seen that in the first round: put `uninitialized` first in the list; this way, the compiler assigns the numeric value zero to it, and that's what the field will be default-initialized to -- which is what you want if one forgets an explicit initializer.